import_export/admin.py

```diff
@@ -535,6 +537,52 @@ def changelist_view(self, request, extra_context=None):
         extra_context["has_import_permission"] = self.has_import_permission(request)
         return super().changelist_view(request, extra_context)
 
+    def _log_actions(self, result, request):
+        """
+        Create appropriate LogEntry instances for the result.
+        """
+        new_rows = list()
+        update_rows = list()
+        delete_rows = list()
+        for row in result:
```

Comment:
I can't figure out quite what it is that smells wrong to me about this block of code - I thought through it a few different ways and I'm just not seeing a super clean way to manage it because of log_actions' interface. If RowResult knew whether it was an ADDITION/CHANGE/DELETION you could at least easily make a dictionary, but then you have a goofy dictionary initialization like:

```
log_entries = {
    ADDITION: [],
    CHANGE: [],
    DELETION: []
}
for row in result:
    log_entries[row.ACTION_TYPE].append(row)
```

Or you have to use defaultdict, which is also kind of ugly.

And I'm not sure that is all that much cleaner lol.

Reply:
Yes, this change to LogEntry interface caused me a bit of a headache, because we have to iterate each type separately now.  I agree it may look off because the same block of code is duplicated 3 times (pretty much).

I'll take another look and see if I can clean it up a bit.

Reply:
@pokken-magic I had a go at tidying this method.  See what you think.

Reply:
Man it's hard to tell but I think this is better? the older I get the more I wonder if "simplifying" code is worth it. But I think this is better :)